sdk/storage/azure-storage-common/test/storage_credential_test.cpp

```diff
@@ -8,8 +8,18 @@ namespace Azure { namespace Storage { namespace Test {
 
   TEST(StorageCredentialTest, DefaultHostCorrect)
   {
-    EXPECT_EQ(Azure::Storage::Details::ParseConnectionString(
-            "DefaultEndpointsProtocol=https;AccountName=testaccount;AccountKey=testkey").BlobServiceUri.GetHost(), "testaccount.blob.core.windows.net");
+    EXPECT_EQ(
+        Azure::Storage::Details::ParseConnectionString(
+            "DefaultEndpointsProtocol=https;AccountName=testaccount;AccountKey=testkey")
+            .BlobServiceUri.GetHost(),
+        "testaccount.blob.core.windows.net");
+
+    EXPECT_EQ(
+        Azure::Storage::Details::ParseConnectionString(
+            "DefaultEndpointsProtocol=https;AccountName=testaccount;AccountKey=testkey;"
+            "EndpointSuffix=core.windows.net")
```

Comment:
We should add a test where the `EndpointSuffix` is not the default value to make sure setting it works.